Tasks/Xcode/xcode.ts

```diff
@@ -316,55 +319,68 @@ async function run() {
                     }
                 }
 
-                if (exportMethod) {
-                    // generate the plist file if we have an exportMethod set from exportOptions = auto or specify
+                if (exportOptions !== 'plist') {
+                    // As long as the user didn't provide a plist, start with an empty one.
+                    // Xcode 7 warns "-exportArchive without -exportOptionsPlist is deprecated"
+                    // Xcode 8+ will error if a plist isn't provided.
                     let plist: string = tl.which('/usr/libexec/PlistBuddy', true);
+
                     exportOptionsPlist = '_XcodeTaskExportOptions.plist';
                     tl.tool(plist).arg(['-c', 'Clear', exportOptionsPlist]).execSync();
-                    tl.tool(plist).arg(['-c', 'Add method string ' + exportMethod, exportOptionsPlist]).execSync();
+
+                    // Add the teamId if provided.
                     if (exportTeamId) {
                         tl.tool(plist).arg(['-c', 'Add teamID string ' + exportTeamId, exportOptionsPlist]).execSync();
                     }
 
-                    if (xcodeVersion >= 9 && exportOptions === 'auto') {
-                        const cloudEntitlement = await sign.getCloudEntitlement(embeddedProvProfiles[0], exportMethod);                        
-                        if (cloudEntitlement) {
-                            tl.debug("Adding cloud entitlement");
-                            tl.tool(plist).arg(['-c', `Add iCloudContainerEnvironment string ${cloudEntitlement}`, exportOptionsPlist]).execSync();
-                        }
-                        let signingOptionForExport = signingOption;
+                    // Add the export method if provided or determined above.
+                    if (exportMethod) {
+                        tl.tool(plist).arg(['-c', 'Add method string ' + exportMethod, exportOptionsPlist]).execSync();
+
+                        if (xcodeVersion >= 9 && exportOptions === 'auto') {
```

Comment:
Does this need to be inside the if(exportMethod) check? Looks like it can be moved outside the if.